pkg/archive/paths.go

```diff
@@ -53,36 +48,22 @@ func WalkZipFiles(ctx context.Context, r *zip.Reader, walkFn FileWalkFn) (err er
 	return nil
 }
 
-// WalkTarFiles accepts an archive path, a TarReaderProvider function for reading the archive, and a walking function to
-// use when traversing files in the archive.
-func WalkTarFiles(ctx context.Context, path string, tarReader TarReaderProvider, walkFn FileWalkFn) (err error) {
-	file, err := os.Open(path)
-	if err != nil {
-		return err
-	}
-	defer func() {
-		if cErr := file.Close(); err == nil && cErr != nil {
-			err = cErr
-		}
-	}()
-	reader, err := tarReader(file)
-	if err != nil {
-		return err
-	}
+// WalkTarFiles walks the files in the provided *tar.Reader, calling walkFn on each header.
+func WalkTarFiles(ctx context.Context, r *tar.Reader, walkFn FileWalkFn) (err error) {
```

Comment:
Same comment as for zip (can either be unexported or possibly even inlined to one calling location)